### PR TITLE
Added letter-spacing support

### PR DESCRIPTION
**Pull request by g-zachar**

### CODE REVIEW

File: src/textures/TextTextureRenderer.mjs [hunk 1]
```diff
@@ -369,8 +380,8 @@ export default class TextTextureRenderer {
             let spaceLeft = wordWrapWidth;
             let words = lines[i].split(' ');
             for (let j = 0; j < words.length; j++) {
-                let wordWidth = this._context.measureText(words[j]).width;
-                let wordWidthWithSpace = wordWidth + this._context.measureText(' ').width;
+                const wordWidth = this.measureText(words[j], letterSpacing);
+                const wordWidthWithSpace = wordWidth + this._context.measureText(' ').width + letterSpacing;
```
Comment (g-zachar) [inline src/textures/TextTextureRenderer.mjs]: You can use `measureText` wrapper you created here.

File: src/textures/TextTextureRenderer.mjs [hunk 2]
```diff
@@ -293,7 +294,17 @@ export default class TextTextureRenderer {
...
-            this._context.fillText(drawLine.text, drawLine.x, drawLine.y);
+
+            if (!letterSpacing) {
+                this._context.fillText(drawLine.text, drawLine.x, drawLine.y);
+            } else {
+                const textSplit = drawLine.text.split('');
+                let x = drawLine.x;
+                for (let i = 0, j = textSplit.length; i < j; i++) {
+                    this._context.fillText(textSplit[i], x, drawLine.y);
+                    x += this._context.measureText(textSplit[i]).width + letterSpacing;
```
Comment (g-zachar) [inline src/textures/TextTextureRenderer.mjs]: You can use `measureText` wrapper you created here.

File: src/textures/TextTextureRenderer.mjs [hunk 3]
```diff
@@ -293,7 +294,17 @@ export default class TextTextureRenderer {
         this._context.fillStyle = StageUtils.getRgbaString(this._settings.textColor);
         for (let i = 0, n = drawLines.length; i < n; i++) {
             let drawLine = drawLines[i];
-            this._context.fillText(drawLine.text, drawLine.x, drawLine.y);
+
+            if (!letterSpacing) {
```
Comment (g-zachar) [inline src/textures/TextTextureRenderer.mjs]: I would prefer `letterSpacing == 0` instead for better readability.

**Comment by erikhaandrikman** [comment]
> Few questions:
> 
> * does it work with negative values as well?
> * is `renderWidth` property updated accordingly?
> 
> Other than that, looks good.

Negative letter-spacing is supported and `renderWidth` will properly output the correct width.

File: src/textures/TextTextureRenderer.mjs (hunk 1 above)
Comment (erikhaandrikman) [reply src/textures/TextTextureRenderer.mjs]: Good call, updated

File: src/textures/TextTextureRenderer.mjs (hunk 3 above)
Comment (erikhaandrikman) [reply src/textures/TextTextureRenderer.mjs]: updated

File: src/textures/TextTextureRenderer.mjs (hunk 2 above)
Comment (erikhaandrikman) [reply src/textures/TextTextureRenderer.mjs]: updated

**Pull request merged**